### PR TITLE
Enable detach annotation for ready nodes

### PR DESCRIPTION
**Pull request by furkatgofurov7**

### CODE REVIEW

**Comment by maelk** [comment]
/test-integration
This would be very useful to prevent Ironic interacting with ready nodes, to enforce the power state for example, when pivoting BMHs that are ready but not in use.

**Review by zaneb — COMMENTED** [review]
Reading back through @hardys's original proposal, I suspect the reason we limited it to those two states was because we were considering having a state machine transistion back to the Unmanaged provisioning state. In the end we went with an operational state change instead, so I can't see any reason not to support this in the Ready state, since that's also a steady state.

/approve

File: controllers/metal3.io/host_state_machine.go [hunk 1]
```diff
@@ -256,7 +256,7 @@ func (hsm *hostStateMachine) checkDetachedHost(info *reconcileInfo) (result acti
 	if hasDetachedAnnotation(hsm.Host) {
 		// Only allow detaching hosts in Provisioned/ExternallyProvisioned states
```
[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply controllers/metal3.io/host_state_machine.go:257]: Thanks for review @zaneb , I will update the comment accordingly in a follow-up PR.

[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply controllers/metal3.io/host_state_machine.go:257]: Follow-up: #967

**Pull request merged**